infrastructure/src/main/java/com/example/opencommerce/infra/pricing/TestPriceBootloader.java

```diff
@@ -0,0 +1,57 @@
+package com.example.opencommerce.infra.pricing;
+
+import com.example.opencommerce.app.pricing.DiscountScenarios;
+import com.example.opencommerce.app.pricing.InitiateNewItemPriceScenario;
+import com.example.opencommerce.app.pricing.commands.ApplyDiscountCommand;
+import com.example.opencommerce.app.pricing.commands.InitiateNewPriceCommand;
+import com.example.opencommerce.domain.ItemId;
+import com.example.opencommerce.domain.Value;
+import com.example.opencommerce.domain.Vat;
+import io.quarkus.arc.Priority;
+import io.quarkus.runtime.StartupEvent;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+import java.time.Instant;
+
+@Slf4j
+@ApplicationScoped
+public class TestPriceBootloader {
+
+    private final InitiateNewItemPriceScenario initiatePrice;
+    private final DiscountScenarios discounts;
+
+    public TestPriceBootloader(InitiateNewItemPriceScenario initiatePrice, DiscountScenarios discounts) {
+        this.initiatePrice = initiatePrice;
+        this.discounts = discounts;
+    }
+
+    void initiateTestPrices(@Observes @Priority(1) StartupEvent ev) {
+        log.info("Populating test price data...");
+        Vat vat = Vat.of(0.23);
+
+        // ItemId 1
+        initiatePrice.initiateNewPrice(new InitiateNewPriceCommand(ItemId.of(1L), Value.of(11.99), vat));
```

Comment:
I'm not sure if this is a good idea. Those objects, yes, it's great to have those. But not for each test. It will take revenge on you, sooner than you think.
The test should be separated, meaning that each test should populate its own data set and clean after itself. That's it for the theory, the practice shows that developers are lazy bastards, so at least the test should clean the tables it will use, populate them with required data, before running the test. Yes, it will take time, but it's a necessary cost, I am afraid. But you gain the ease of creating new tests, new features. Worth to take that cost.